dcm4chee-arc-retrieve/src/main/java/org/dcm4chee/arc/retrieve/impl/RetrieveContextImpl.java

```diff
@@ -195,20 +196,25 @@ public ArchiveAEExtension getArchiveAEExtension() {
 
     @Override
     public String[] getAccessControlIDs() {
-        String [] accessControlIDs = arcAE.getAccessControlIDs();
+        Set<String> accessControlIDSet = new HashSet<>(Arrays.asList(arcAE.getAccessControlIDs()));
+        Set<String> httpRequestAccessControlIDset = ClientRoles.get(this.httpServletRequestInfo);
+        if(httpRequestAccessControlIDset != null)
+            accessControlIDSet.addAll(httpRequestAccessControlIDset);
+
         if (null != requestAssociation)
         {
             AAssociateAC ac = requestAssociation.getAAssociateAC();
             if (null != ac) {
                 UserIdentityAC userIdentityAC = ac.getUserIdentityAC();
+
                 if (userIdentityAC instanceof ArchiveUserIdentityAC) {
-                    Set<String> set = new HashSet<>(Arrays.asList(accessControlIDs));
-                    ((ArchiveUserIdentityAC) userIdentityAC).filterRolesByClientRoles(set);
```

Comment:
I'm no longer calling `filterRolesByClientRoles` as I'm not sure of its purpose, but removing it probably breaks something. I am just combining the three sets of acess control IDs - from AET, from the DICOM user identity, and from the http token.